extras/devices/LVFS/LVFS.py

```diff
@@ -0,0 +1,301 @@
+            try:
+                expected_ver = dev["targetVersion"]
+            except KeyError:
+                continue
+            rc, stdout, stderr = self.run_cmd(
+                f"sudo fwupdmgr get-results {dev['DeviceId']} --json"
+            )
+            get_results = json.loads(stdout)
+            logger.debug(f"$fwupdmgr get-result = \n{stdout}")
+            try:
+                new_fw = get_results["Releases"][0]
+                update_state = get_results["UpdateState"]
+            except KeyError:
+                msg = f"[{dev_name}] unable to determine if new firmware is landed due to missing results"
+                logger.info(msg)
+                print(msg)
+                fwupd_result = False
+                continue
+
+            if new_fw["Version"] == expected_ver and update_state == 2:
+                msg = f"[{dev_name}] firmware flashed {dev['Version']} → {expected_ver}"
+            else:
+                FwupdUpdateState = [
+                    "FWUPD_UPDATE_STATE_UNKNOWN",
+                    "FWUPD_UPDATE_STATE_PENDING",
+                    "FWUPD_UPDATE_STATE_SUCCESS",
+                    "FWUPD_UPDATE_STATE_FAILED",
+                    "FWUPD_UPDATE_STATE_NEEDS_REBOOT",
+                    "FWUPD_UPDATE_STATE_FAILED_TRANSIENT",
+                    "FWUPD_UPDATE_STATE_LAST",
+                ]
+                msg = f"[{dev_name}] firmware upgrade failed. {FwupdUpdateState[update_state]}"
+                fwupd_result = False
+            print(msg)
+            logger.info(msg)
+
+        return fwupd_result
+
+    def check_connectable(self, timeout):
+        """
+        After DUT reboot, check if SSH to DUT works within a given timeout period
+
+        :param timeout: wait time for regaining DUT access
+        :type timeout: int
+        """
+        logger.info(f"check and wait for {timeout}s until SSH is connectable")
+        status = "1"
+        timeout_start = time.time()
+
+        while status != "0" and time.time() < timeout_start + timeout:
+            status = subprocess.check_output(
+                f"timeout 10 ssh {SSH_OPTS} {self.user}@{self.ipaddr} /bin/true 2>/dev/null; echo $?",
+                shell=True,
+                universal_newlines=True,
+            ).strip()
+        if status != "0":
+            logger.error(f"Failed to SSH to {self.ipaddr} after {timeout}s")
+            raise RuntimeError(
+                f"Failed to SSH to {self.ipaddr} after {timeout}s"
+            )
+        delta = time.time() - timeout_start
+        logger.info(f"{self.ipaddr} is SSHable after {int(delta)}s")
+
+    def reboot(self):
+        logger.info("reboot DUT")
+        self.run_cmd("sudo reboot", raise_stderr=False)
+        time.sleep(10)
+        self.check_connectable(self.reboot_timeout)
+
+
+class LenovoNB(LVFSDevice):
+    fw_update_type = "LVFS-ext"
+    vendor = "LENOVO"
+
+
+# class HPNB(LVFSDevice):
+# class DellNB(LVFSDevice)
```

Comment:
It was there intentionally for the future implementation. But maybe it's better to remove them now since there's already a template class LenovoNB. Removed.